src/components/Form.js

```diff
@@ -20,10 +20,13 @@ const Form = () => {
     e.preventDefault();
     const data = {
       ...book,
+      category: 'uncategorized',
       id: Date.now().toString(),
     };
-    if (book.title.length > 0 && book.title.length > 0) {
-      dispatch(addBookAction(data));
+    if (book.title.length > 0 && book.author.length > 0) {
+      dispatch(addBookAction(data)).then(() => {
+        dispatch(addBook(data));
+      });
```

Comment:
- [ ] When I add a book, it is not persisted through the API endpoint. I get this error in the console:

![image](https://user-images.githubusercontent.com/77106962/208266415-8f8599f2-c635-4f94-a286-98ae2de319f8.png)

Although the book is persisted in the redux store, it is not persisted through the endpoint, so once I reload the page, the data is gone. Kindly fix it 👍

Reply:
@Gambit142 I have tried so many things. I have no idea why it isn't persisting in the API

Reply:
@lukman155, try checking the API documentation for the parameters that are needed to make a successful POST request and ensure that you attaching the right parameters. You can also check for syntax errors. 👍